crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -713,47 +725,35 @@ where
     fn latest_valid_hash_for_invalid_payload(
         &mut self,
         parent_hash: B256,
-        insert_err: Option<&InsertBlockErrorKind>,
-    ) -> Option<B256> {
-        // check pre merge block error
-        if insert_err.map(|err| err.is_block_pre_merge()).unwrap_or_default() {
-            return Some(B256::ZERO)
+    ) -> ProviderResult<Option<B256>> {
+        // Check if parent exists in side chain or in canonical chain.
+        if self.blockchain.find_block_by_hash(parent_hash, BlockSource::Any)?.is_some() {
+            return Ok(Some(parent_hash))
         }
```

Comment:
`insert_err` was present only during single invocation, the check was moved there